Purchases/Public/RCPurchases.m

```diff
@@ -223,13 +266,71 @@ - (void)setDelegate:(id<RCPurchasesDelegate>)delegate
 #pragma mark - Public Methods
 
 #pragma mark Attribution
+
+- (void)addAttributionData:(NSDictionary *)data
+               fromNetwork:(RCAttributionNetwork)network
+{
+    [self addAttributionData:data fromNetwork:network forNetworkUserId:nil];
+}
+
 - (void)addAttributionData:(NSDictionary *)data
                fromNetwork:(RCAttributionNetwork)network
+          forNetworkUserId:(NSString * _Nullable)networkUserId
 {
-    if (data.count > 0) {
-        [self.backend postAttributionData:data
-                              fromNetwork:network
-                             forAppUserID:self.appUserID];
+    NSString *networkKey = [NSString stringWithFormat:@"%ld",(long)network];
+    NSString *advertisingIdentifier = [self.attributionFetcher advertisingIdentifier];
+    NSString *newValueForNetwork = [NSString stringWithFormat:@"%@_%@", advertisingIdentifier, networkUserId];
```

Comment:
I would set this closer to where it's used since it doesn't seem to be needed at this scope level.